### PR TITLE
Feature/first importance insight notification

### PR DESCRIPTION
**Pull request by shalom938**

### CODE REVIEW

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/notifications/InsightsNotificationStartupActivity.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.digma.intellij.plugin.notifications
+
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.StartupActivity
+
+class InsightsNotificationStartupActivity: StartupActivity {
+    override fun runActivity(project: Project) {
+        project.service<InsightsNotificationsService>().waitForEvents()
```
Comment (arik-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/notifications/InsightsNotificationStartupActivity.kt:9]: the name `waitForEvents` sounds blocking. maybe just name it `startWorking`

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/notifications/InsightsNotificationsService.kt [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    fun waitForEvents() {
+
+        Log.log(logger::info, "starting insights notification service")
+
+        @Suppress("UnstableApiUsage")
+        DisposingScope(this).launch {
+
+            while (true) {
+
+                delay(60000)
```
Comment (arik-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/notifications/InsightsNotificationsService.kt:51]: ohhh, it is blocking... why is that?

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/notifications/InsightsNotificationsService.kt [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+    private fun showNotificationForDurationChangeEvent(durationChangedEvent: CodeObjectDurationChangeEvent) {
+        var codeObjectId = durationChangedEvent.codeObjectId
+        if (durationChangedEvent.codeObjectId == null) {
+            codeObjectId = durationChangedEvent.spanCodeObjectId
+        }
+
+        if (codeObjectId != null) {
+
+            showInsightNotification(
```
Comment (arik-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/notifications/InsightsNotificationsService.kt:121]: `CodeObjectDurationChangeEvent` is not necessarily insight notification

File: model/src/main/kotlin/org/digma/intellij/plugin/model/rest/event/LatestCodeObjectEventsRequest.kt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+package org.digma.intellij.plugin.model.rest.event
+
+data class LatestCodeObjectEventsRequest(val environments: List<String>,val fromDate: String)
```
Comment (arik-dig) [inline model/src/main/kotlin/org/digma/intellij/plugin/model/rest/event/LatestCodeObjectEventsRequest.kt:3]: fromDate should be `DateTime`
```suggestion
data class LatestCodeObjectEventsRequest(val environments: List<String>,val fromDate: DateTime)
```

File: model/src/main/kotlin/org/digma/intellij/plugin/persistence/PersistenceData.kt [hunk 5]
```diff
@@ -17,4 +17,5 @@ data class PersistenceData(
         var isFirstTimePluginLoaded: Boolean = false,
         var userId: String? = null,
         var firstTimePerformanceMetrics: Boolean = false,
+        var lastInsightsEventTime: String? = null,
```
Comment (arik-dig) [inline model/src/main/kotlin/org/digma/intellij/plugin/persistence/PersistenceData.kt:20]:
```suggestion
        // stored as ZonedDateTime
        var lastInsightsEventTime: String? = null,
```

**Review by arik-dig — APPROVED** [review]
please modify signatures to `DateTime`

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/notifications/InsightsNotificationsService.kt (hunk 2 above)
Comment (shalom938) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/notifications/InsightsNotificationsService.kt:51]: its not blocking its a coroutine

File: ide-common/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsService.java [hunk 6]
```diff
@@ -212,6 +214,12 @@ public List<GlobalInsight> getGlobalInsights() throws AnalyticsServiceException
         return insights;
     }
 
+
+    public LatestCodeObjectEventsResponse getLatestEvents(@NotNull String lastReceivedTime) throws AnalyticsServiceException{
```
[the remark this answers is not in the document]
Comment (shalom938) [reply ide-common/src/main/java/org/digma/intellij/plugin/analytics/AnalyticsService.java:218]: done

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/notifications/InsightsNotificationStartupActivity.kt (hunk 1 above)
Comment (shalom938) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/notifications/InsightsNotificationStartupActivity.kt:9]: I'll keep it, its a background thread

**Review by doppleware — APPROVED** [review]

**Pull request merged**